ignite3/src/main/java/site/ycsb/db/ignite3/AbstractSqlClient.java

```diff
@@ -1,32 +1,110 @@
 package site.ycsb.db.ignite3;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
 import site.ycsb.ByteIterator;
+import site.ycsb.DBException;
 
 abstract class AbstractSqlClient extends IgniteAbstractClient {
+  /** Prepared statement for reading values. */
+  protected static ThreadLocal<PreparedStatement> readPreparedStatement;
 
-  static String prepareInsertStatement(String table, String key, Map<String, ByteIterator> values) {
+  /** Prepared statement for inserting values. */
+  protected static ThreadLocal<PreparedStatement> insertPreparedStatement;
+
+  /** {@inheritDoc} */
+  @Override
+  public void init() throws DBException {
+    super.init();
+
+    readPreparedStatement = new ThreadLocal<>();
+    insertPreparedStatement = new ThreadLocal<>();
```

Comment:
Every client thread calls `init()` so it would be better to either assign a value upon variable declaration or move these two lines under a critical section.